File: dynabuffers-python/tests/usecase/Schema15Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema15Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(f"{self.root_dir}/schema15.dbs"))

    def test_resolve_union_type_with_ordinal(self):
        obj = {"name": "test", "content": bytearray(b"text"), ":type": 1}
        result = self.engine.deserialize(self.engine.serialize(obj))
        self.assertEqual(result, obj)

    def test_resolve_union_type_with_qualifier(self):
        obj = {"name": "test", "content": bytearray(b"text"), ":type": "ByteArrayData"}
        result = self.engine.deserialize(self.engine.serialize(obj))
        self.assertEqual(result, {"name": "test", "content": bytearray(b"text"), ":type": 1})


if __name__ == "__main__":
    unittest.main()

File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema15Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class Schema15Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema15Test::class.java.getResourceAsStream("/schema15.dbs"))
    }

    @Test
    fun resolveUnionTypeWithOrdinal() {
        val message = mapOf(
                "name" to "test",
                "content" to "text".toByteArray(),
                ":type" to 1
                )
        val result = engine.deserialize(engine.serialize(message))

        assertMap(message, result)
    }

    @Test
    fun resolveUnionTypeWithQualifier() {
        val message = mapOf(
                "name" to "test",
                "content" to "text".toByteArray(),
                ":type" to "ByteArrayData"
        )
        val result = engine.deserialize(engine.serialize(message))

        assertMap(mapOf(
                "name" to "test",
                "content" to "text".toByteArray(),
                ":type" to 1
        ), result)
    }

}
